agentlightning/store/client_server.py

```diff
@@ -1371,6 +1247,9 @@ def __init__(
         self._dequeue_was_successful: bool = False
         self._dequeue_first_unsuccessful: bool = True
 
+        # Track server shutdown state to handle errors gracefully
+        self._server_shutting_down: bool = False
```

Comment:
How about having a  `_server_online` shared flag and we can sunset hacky flags like `_dequeue_first_unsuccessful`.